open-ce/test_feedstock.py

```diff
@@ -0,0 +1,223 @@
+            print("--------------------------------")
+            print("Failed test: " + self.name)
+            print(self.output)
+            print("--------------------------------")
+
+    def failed(self):
+        """
+        Returns whether or not a test failed.
+        """
+        return self.returncode != 0
+
+def load_test_file(test_file):
+    """
+    Load a given test file.
+
+    Args:
+        test_file (str): Path to the test file to load.
+    """
+    with open(test_file, 'r') as stream:
+        test_file_data = yaml.safe_load(stream)
+
+    return test_file_data
+
+def gen_test_commands(test_file=utils.DEFAULT_TEST_CONFIG_FILE):
+    """
+    Generate a list of test commands from the provided test file.
+
+    Args:
+        test_file (str): Path to the test file.
+    """
+    test_data = load_test_file(test_file)
+
+    time_stamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+    conda_env = "open-ce-conda-env-" + time_stamp
+
+    test_commands = []
+
+    # Create conda environment for testing
+    test_commands.append(TestCommand(name="Create conda environment " + conda_env,
+                                     conda_env=conda_env,
+                                     create_env=True))
+
+    for test in test_data['tests']:
+        test_commands.append(TestCommand(name=test.get('name'), conda_env=conda_env, bash_command=test.get('command')))
+
+    test_commands.append(TestCommand(name="Remove conda environment " + conda_env,
+                                     conda_env=conda_env,
+                                     clean_env=True))
+
+    return test_commands
+
+def run_test_commands(conda_env_file, test_commands):
+    """
+    Run a list of tests within a conda environment.
+
+    Args:
+        conda_env_file (str): The name of the conda environment file used to create the conda environment.
+        test_commands (:obj:`list` of :obj:`TestCommand): List of test commands to run.
+    """
+    failed_tests = []
+    for test_command in test_commands:
+        test_result = test_command.run(conda_env_file)
+        if test_result.failed():
+            failed_tests.append(test_result)
+
+    return failed_tests
+
+def display_failed_tests(failed_tests):
+    """
+    Display a list of failed tests.
+
+    Args:
+        failed_tests (:obj:`list` of :obj:`TestResult`): A list of failed TestResult's.
+    """
+    # If any collected output, return 1, else return 0
+    if failed_tests:
+        for failed_test in failed_tests:
+            failed_test.display_failed()
+        print("The following tests failed: " + str([failed_test.name for failed_test in failed_tests]))
+    else:
+        print("All tests passed!")
+
+def make_parser():
+    ''' Parser for input arguments '''
+    arguments = [utils.Argument.CONDA_ENV_FILE]
+    parser = utils.make_parser(arguments, description = 'Test a feedstock as part of Open-CE')
+
+    return parser
+
+def test_feedstock(arg_strings=None):
```

Comment:
Naming nit: Should we be using `test_package` here (and in the name of the file) since it really is a test on the package content and not the feedstock itself?

Reply:
I'm not sure. It might even make sense to be test_repository, since these tests don't necessarily need to be associated with a particular package. I went with feedstock here since that is what we changed the name of "repository" in the Open-CE environment files to "feedstock".

I'd say what this script does is: "Run every test within a repository (or directory I suppose) inside of a given conda environment"

Reply:
OK. We do use `feedstock` in a bit broader context already so this is fine.